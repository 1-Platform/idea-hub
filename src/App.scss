@import '@patternfly/react-styles/css/utilities/Spacing/spacing.css',
  '@patternfly/react-styles/css/utilities/Text/text.css',
  '@patternfly/react-styles/css/utilities/Alignment/alignment.css',
  '@patternfly/react-styles/css/utilities/Sizing/sizing.css';

html,
body,
#root {
  width: 100%;
  padding: 0;
  margin: 0;
}

a {
  text-decoration: none;
  color: unset;
}

.pf-c-page__main {
  overflow: unset;
}

.pf-c-button.pf-m-primary {
  --pf-c-button--m-primary--BackgroundColor: #61bffa;
  padding: 12px 16px;
}

.pf-c-button.pf-m-secondary {
  --pf-c-button--after--BorderColor: #61bffa;
  --pf-c-button--m-secondary--Color: #61bffa;
}

.pf-c-button.pf-m-link {
  --pf-c-button--m-link--Color: 'unset';
}

.pf-c-tabs__item.pf-m-current button span {
  font-weight: 600;
}

.pf-c-page {
  --pf-c-page--BackgroundColor: #fff;
}

.pf-c-tabs__item.pf-m-current {
  --pf-c-tabs__link--after--BorderColor: rgba(120, 200, 250);
  --pf-c-tabs__link--Color: #000000;
}

.pf-c-tabs {
  --pf-c-tabs--before--BorderBottomWidth: 0;
  &__link:hover {
    --pf-c-tabs--before--BorderBottomWidth: 0;
  }

  &:hover {
    --pf-c-tabs--before--BorderBottomWidth: 0;
  }
}

.pf-c-chip {
  --pf-c-chip--BackgroundColor: #eeeeee;
  --pf-c-chip--BorderRadius: 16px;
  padding: 4px 12px;
}

.rounded {
  border-radius: 8px;
}
