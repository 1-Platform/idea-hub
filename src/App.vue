<template>
  <div class="container" id="app">
    <Header />
    <main class="pf-l-grid pf-m-gutter">
      <router-view />
    </main>
    <Footer />
  </div>
</template>

<style>
html,
body {
  font-family: "Red Hat Text", sans-serif;
  height: 100%;
  width: 100%;
  padding: 0;
  margin: 0;
}

.container {
  height: 100vh;
  display: flex;
  flex-direction: column;
}

main {
  flex: 1;
}
/**
*  Media queries here
*/
/* Small devices (landscape phones, 576px and up) */
@media (min-width: 576px) {
  .container {
    max-width: 540px;
    margin: auto;
  }
}

/* Medium devices (tablets, 768px and up) The navbar toggle appears at this breakpoint */
@media (min-width: 768px) {
  .container {
    max-width: 720px;
    margin: auto;
  }
}

/* Large devices (desktops, 992px and up) */
@media (min-width: 992px) {
  .container {
    max-width: 960px;
    margin: auto;
  }
}

/* Extra large devices (large desktops, 1200px and up) */
@media (min-width: 1200px) {
  .container {
    max-width: 1140px;
    margin: auto;
  }
}
</style>

<script lang="ts">
import { Options, Vue } from "vue-class-component";
import Header from "@/views/Header.vue"; // @ is an alias to /src
import Footer from "@/views/Footer.vue"; // @ is an alias to /src

@Options({
  components: {
    Header,
    Footer,
  },
})
export default class Home extends Vue {}
</script>