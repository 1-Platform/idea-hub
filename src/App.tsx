import './App.css';

const App = (): JSX.Element => {
    return <div>React + Typescript</div>;
};

export default App;
