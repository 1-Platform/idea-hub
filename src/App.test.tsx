import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from './App';

afterEach(() => {
  cleanup();
});

test('testing main app renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Root App matches the snapshot saved', function () {
  const renderer = ShallowRenderer.createRenderer();
  const tree = renderer.render(<App />);
  expect(tree).toMatchSnapshot();
});
