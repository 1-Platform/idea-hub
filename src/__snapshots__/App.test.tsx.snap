// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root App matches the snapshot saved 1`] = `
<BrowserRouter>
  <Router />
</BrowserRouter>
`;
